async-aggregators/src/utils/q_custom_slider.py

```diff
@@ -0,0 +1,133 @@
+import typing
+
+from PyQt6.QtCore import QPoint, QRect, Qt
+from PyQt6.QtGui import QMouseEvent, QPainter
+from PyQt6.QtWidgets import (
+    QHBoxLayout, QSlider, QStyle, QStyleOptionSlider, QVBoxLayout, QWidget
+)
+
+
+class QCustomSlider(QWidget):
+    """A custom slider with tick labels"""
+
+    def __init__(self, minimum: int, maximum: int, interval: int = 1,
+                 orientation: Qt.Orientation = Qt.Orientation.Horizontal,
+                 labels: list | None = None, p0: typing.Any = 0, parent: typing.Any = None,
+                 suppress_mouse_move: bool = False) -> None:
+        super().__init__(parent=parent)
+
+        levels = range(minimum, maximum + interval, interval)
+
+        if labels is not None:
+            if not isinstance(labels, (tuple, list)):
+                raise Exception("<labels> is a list or tuple.")
+            if len(labels) != len(levels):
+                raise Exception("Size of <labels> doesn't match levels.")
+            self.levels = list(zip(levels, labels))
+        else:
+            self.levels = list(zip(levels, map(str, levels)))
```

Comment:
This version is more lenient on what types it will accept for `labels`, which could be good or bad. In general though, it's best to embrace the duck-typing and let the user pass whatever they want as long as it works. This version also has fewer branches, making it easier to read.
```suggestion
        if labels is None:
            labels = map(str, levels)
        labels = list(labels)
        if len(labels) != len(levels):
            raise ValueError("Size of <labels> doesn't match levels.")
        try:
            self.levels = list(zip(levels, labels))
        except TypeError:
            raise TypeError("<labels> is not iterable.")
```